src/factories.js

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import components from './index';
+
+let factories = {};
+
+Object.keys(components).forEach(function (name) {
+  if (components[name] instanceof React.Component.constructor) {
```

Comment:
Does this work with components created with both `React.createClass` and ES6 style `class NAME extends React.Component`? To be safe I think it would be a good idea to extract this logic to a function and write some tests against it.